git-tutorials/git.md

```diff
@@ -0,0 +1,55 @@
+- [Getting Started with Git/GitHub](#getting-started-with-git-github)
+    + [Clone](#clone)
+    + [Pull](#pull)
+    + [Commit](#commit)
+    + [Push](#push)
+    + [Pull Request](#pull-request)
+
+# Getting Started with Git/GitHub
+To use Git, you'll need to download it [here](https://git-scm.com/downloads).
+
+### Clone
+First things first, we'll want to get a copy of our repository on our computer locally by cloning our repository. We will need the HTTPS address on the repository page which you can copy/paste from the browser. We will now use our terminal's shell commands to move to the place where we want to put our directory copy.
+
+For example, if we wanted to change to our Documents directory, we would do the following:
+`cd Documents`
+
+To clone our repository, we use the following command in our terminal:
+`git clone <https address here>`
+
+For example:
+`git clone https://github.com/ThinkFiveable/open fiveable-open`
+
+Again, this will create a local copy of the repository at the directory specified earlier. You will then need to open these files in your preferred text editor to make your changes.
+ 
+### Pull
+You will need to use `git pull` to fetch the latest changes from another repository or local branch. When you run `git pull` in your terminal, remote content will be downloaded and immediately change the local state of your repository to match the new content. It is strongly recommended to not `git pull` when you have uncommitted changes. 
+
+In the event that you do have uncommitted changes and want to `git pull`, you will need to do the following in your command line.
+First, `git stash` which "stashes" your changes away temporarily. Then, `git pull` to fetch the latest changes. Next, `git stash pop` to apply the changes back onto your local copy.
```

Comment:
```suggestion
First, `git stash` which "stashes" your changes away temporarily. Then, `git pull` to fetch the latest changes. Next, `git stash apply` to apply the changes back onto your local copy, and then `git stash drop` to remove those changes from the stash stack.
```
https://codingkilledthecat.wordpress.com/2012/04/27/git-stash-pop-considered-harmful/.

In short, unexpected behavior can occur in the event of pull/merge conflicts, which means git stash pop won't actually remove from the stack like expected. IMO it'd be better to be consistent to begin with than to have to teach edge cases like that.